js/bulkhead-callback.js

```diff
@@ -279,16 +283,73 @@ var bulkheadCallBack = (function() {
         return match;
     };
 
+    var __validateEditorContent_BulkheadStep = function(content) {
+        var match = false;
+        try {
+            var pattern = "@Bulkhead\\(50\\)\\s*public Service serviceForVFA";
+            var regExpToMatch = new RegExp(pattern, "g");
+            content.match(regExpToMatch)[0];
+            match = true;
+        } catch (ex) {
+
+        }
+        return match;
+    };
+
+    var __validateEditorContent_AsyncBulkheadStep = function(content) {
+        var match = false;
+        try {
+            var pattern1 = "@Asynchronous\\s*@Bulkhead\\(value\\s?=\\s?50,\\s*" + 
+                "waitingTaskQueue\\s?=\\s?50\\)\\s*" +
+                "public Future<Service> serviceForVFA";
```

Comment:
You need a readonly beginning and ending boundaries to make sure the code block changes are in the right place. I wonder whether it is better to check the entire content in one regex rather than breaking down to 3 here.

Reply:
I think for now we can keep it broken down in 3. The instruction seems really large with 4-5 actions, there could be a possibility of splitting up the instruction. If it's not needed, it's very easy to combine all 3 later on.